pkg/cgroups/cgroups.go

```diff
@@ -763,3 +764,84 @@ func HostCgroupRoot() (string, error) {
 	)
 	return "", fmt.Errorf("failed to set cgroup root: %w", err)
 }
+
+// CgroupIDFromPID returns the cgroup id for a given pid.
+func CgroupIDFromPID(pid uint32) (uint64, error) {
+	cgroupFile := fmt.Sprintf("%s/%d/cgroup", option.Config.ProcFS, pid)
+	data, err := os.ReadFile(cgroupFile)
+	if err != nil {
+		return 0, err
+	}
+
+	pathPrefix := fmt.Sprintf("%s/1/root/sys/fs/cgroup", option.Config.ProcFS)
+
+	var path string
+	v2Prefix := "0::"
+	v1Prefix := fmt.Sprintf("%d:%s:", GetCgrpSubsystemIdx(), GetCgrpControllerName())
```

Comment:
Why not call here if:  `GetCgroupMode() == CGROUP_UNIFIED`  assume only cgroupv2 and you will have one line to parse.

For everything else what you are already doing and it will be definitely under cgroupv1, even if systemd adds the cgroupv2 hierarchy, we don't use it, we follow the default k8s used one cgroupv1. So yeh looking for that:
`v1Prefix := fmt.Sprintf("%d:%s:", GetCgrpSubsystemIdx(), GetCgrpControllerName())`   will work, prefixing it later with pathprefix + controller name and the path.

Reply:
> Why not call here if: GetCgroupMode() == CGROUP_UNIFIED assume only cgroupv2 and you will have one line to parse.

Haven't thought of that but, IMO, it makes the code more complicated so leaving things as is seems like the best option to me.

Reply:
yeh the problem is could be theoretical one has to check kernel sources, but there are hybrid installation where both are available, systemd mounts at it /sys/fs/cgroup/unified/ for cgroupv2 features and not resources management, now I can't recall how they show up inside the /proc/pid/cgroup or if it a named cgroup hierarchy, if they are by order, etc... I forget. But the point you may multiple lines there, so if we are checking that better improve it and separate it by:
1. if unified mode you only lookup for v2prefix
2. anything else you lookup for v1prefix

This way you don't parse them both at same time...

Reply:
Looks perfect and nice style BTW ;-) !

Reply:
Thanks for the review!